crate_universe/3rdparty/crates/BUILD.ahash-0.7.6.bazel

```diff
@@ -245,6 +245,6 @@ cargo_build_script(
 
 alias(
     name = "build_script_build",
-    actual = "ahash_build_script",
+    actual = "//:ahash_build_script",
```

Comment:
Hmm another issue, `Label::from_str(":foo").unwrap().repr() == "//:foo"` which is not equivalent.  Same issue with a plain `foo` instead of `:foo`.

Reply:
Seems like making emitting `//` conditional on either `repository` or `package` being present would work.  I have mixed feelings on whether that's "correct" though.

Reply:
Yeah, this is definitely not correct as-is, and the conditional we want isn't on what the label looks like, it should be conditional on the relationship between where the label was parsed and where the label is being serialized - there are really two different meanings here - there's a concept of a "relative" label and an "absolute" label.

We need to be aware of this at parse time. I can imagine two ways of handling this reasonably:
1. Change `Label` to be something like:
```rust
struct Label {
    package: PackageOrRelative,
    target: String,
}

enum PackageOrRelative {
    Package {
        repository: String,
        package: String,
    },
    Relative,
}
```
and at render time we output the `//` if `PackageOrRelative` is a `Package`, and omit it if it's a `Relative`. This is a little sketchy, as "label parse time" isn't really a time when we know whether we're going to serialize the label in the same package or a different one, but to simplify things we can make the assumption that the caller knows.

2. At parse time, pass in the package a label is being parsed relative to (and change `package` from an `Option<String>` to a `String` - every label _is_ in a package so this shouldn't _really_ be an `Option` (but I can believe that at existing use sites, e.g. in the `FromStr` implementation we don't _know_ the package we're parsing from). Then we would unconditionally serialize the `//` because we'd always know the package we were in. If we do this, there's an interesting question as to whether at serialisation time we want to accept "the package being serialised into" so that we can output a relative label if the package happens to match.

All of which goes back to the "how many label implementations" question we have. There are some really different use-cases for parsing/serialising a label - "This is just a typed string" lends itself to our current implementation (and leans towards option 1). "This is a type which can be used in different contexts to decide e.g. whether a package should be serialised or not based on where it's used" is more complex to use (as we need to understand parsing and serializing contexts) but may allow solving more general problems (and leans towards option 2). I'm generally fine with us having two types - one which is "basically a newtype around a string" and one which is aware of packages, but also happy for us to end up with one implementation as long as it's reasonably ergonomic...

I'd probably lean towards 1 here, because there isn't really a correct way for us to implement `FromStr`, `Serialize`, and `Deserialize` with option 2...